File: src/main/java/com/lxy/flowplan/pojo/Project.java
package com.lxy.flowplan.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("project")
public class Project {

    // Project 是排期的根对象；Task、Plan、Setting 都围绕它展开。
    @TableId(type = IdType.AUTO)
    private Integer id;

    @JsonProperty("user_id")
    private Integer userId;

    private String name;

    private String description;

    @JsonProperty("begin_date")
    private LocalDate beginDate;

    @JsonProperty("finish_date")
    private LocalDate finishDate;

    private LocalDate deadline;

    private String status = "NOT_STARTED";

    @JsonProperty("risk_level")
    private String riskLevel = "OK";

    // 上游字段变化后置为 true，提示计划模块需要重新生成后续安排。
    @JsonProperty("need_replan")
    private Boolean needReplan = false;

    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/api/project.js
import request from "./request";

function normalizeProject(project) {
    return {
        ...project,
        status: project.status || "NOT_STARTED",
        risk_level: project.risk_level || "OK",
        need_replan: Boolean(project.need_replan)
    };
}

function normalizeProjectPayload(project) {
    return {
        name: project.name?.trim(),
        description: project.description?.trim() || null,
        begin_date: project.begin_date || null,
        deadline: project.deadline || null
    };
}

export async function listProjects() {
    const projects = await request.get("/project/list");
    return Array.isArray(projects) ? projects.map(normalizeProject) : [];
}

export function addProject(project) {
    return request.post("/project/add", normalizeProjectPayload(project));
}

export function updateProject(id, project) {
    return request.put(`/project/${id}`, normalizeProjectPayload(project));
}

export function deleteProject(id) {
    return request.delete(`/project/${id}`);
}
